internal/provider/resource_gitlab_project.go

```diff
@@ -671,7 +671,16 @@ branch using a ` + "`DELETE`" + ` request. Then define the desired branch protec
 		Importer: &schema.ResourceImporter{
 			StateContext: schema.ImportStatePassthroughContext,
 		},
-		Schema: resourceGitLabProjectSchema,
+		Schema: constructSchema(resourceGitLabProjectSchema, map[string]*schema.Schema{
+			"skip_wait_for_default_branch_protection": {
+				Description: `If ` + "`true`" + `, the default behavior to wait for the default branch protection to be created is skipped.
+This is necessary if the current user is not an admin and the default branch protection is disabled on an instance-level.
+There is currently no known way to determine if the default branch protection is disabled on an instance-level for non-admin users.`,
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
```

Comment:
@timofurrer  - Technically setting this to false creates a breaking change. Configs that would previously work will now stop working. Do we need to set this to `true` for backwards compatibility?

Reply:
Resolved with inverting the boolean check.

Reply:
hmmm .. can you elaborate? We previously had this https://github.com/gitlabhq/terraform-provider-gitlab/issues/1127 which caused a timeout (which is now the case with a non-admin token and this set to `false`).